lib/pbench/server/api/resources/api_key.py

```diff
@@ -140,7 +148,7 @@ def _delete(
             Success: 200
 
         Raises:
-            APIAbort, reporting "UNAUTHORIZED" and "FORBIDDEN"
+            APIAbort, reporting "UNAUTHORIZED", "NOT_FOUND" and "FORBIDDEN"
```

Comment:
GitHub won't let me comment where I *want* to comment on this, but note that if with your existing `APIKey.query` you could do `key = APIKey.query(id=key_id, user=user)` and `key` will be `None` if that key ID isn't owned by the authenticated user, which would simplify your logic a bit. (Or, if you fix `query` to return `.all()`, it'd be `[]` instead of `None`, which is equally easy to deal with and in fact pretty much transparent if you just do `if key:`. 😄 And, yes, it'd be slightly better if the API didn't allow the caller to know whether the key ID was bad or simply wasn't owned by them.)